### PR TITLE
Fixes #34186 - consume Pagination from core

### PR DESCRIPTION
**Pull request by Ron-Lavi**
#### What are the changes introduced in this pull request?
Refactor to use the PF4 pagination from core
#### Considerations taken when implementing this change?
Should be tested with https://github.com/theforeman/foreman/pull/8990
#### What are the testing steps for this pull request?
Checkout the PR branch from core and see if the pagination still works as expected and in some pages, the PF3 pagination should be replaced with the new implementation

### CODE REVIEW

File: .github/workflows/react_tests.yml [hunk 1]
```diff
@@ -8,33 +8,32 @@ defaults:
...
+          node-version: '12.x'
+      # We could update the postinstall action for foreman to look for an environment variable for plugin webpack dirs
+      # before kicking off the ruby script to find them, this would eliminate the ruby dep and running `npm i` in Katello.
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: 2.6
+      - name: Checkout Foreman repo
+        uses: actions/checkout@v2
+        with:
+          repository: Ron-Lavi/foreman
```
Comment (Andrewgdewar) [inline .github/workflows/react_tests.yml:28]: was this commit intentional?

Comment (Ron-Lavi) [reply .github/workflows/react_tests.yml:28]: yes, this way i can test it with the core PR (my branch)

**Comment by Ron-Lavi** [comment]
@Andrewgdewar @jeremylenz what's going on with the tests? :P
can you help here?

**Comment by jeremylenz** [comment]
[test katello]

**Comment by jeremylenz** [comment]
@Ron-Lavi I think the katello errors are related too:
```
ModuleNotFoundError: Module not found: Error: Can't resolve 'foremanReact/components/Pagination' in '/home/jenkins/workspace/katello-pr-test/webpack/components/Table'
```
it seems like foremanReact is broken..

**Comment by jeremylenz** [comment]
I get similar errors when just trying to run the server locally with this checked out:
```
19:21:19 webpack.1 | ERROR in ../katello/webpack/components/Table/PageControls.js
19:21:19 webpack.1 | Module not found: Error: Can't resolve 'foremanReact/components/Pagination' in '/home/vagrant/katello/webpack/components/Table'
19:21:19 webpack.1 | 
19:21:19 webpack.1 | ERROR in ../katello/webpack/components/pf3Table/components/Table.js
19:21:19 webpack.1 | Module not found: Error: Can't resolve 'foremanReact/components/Pagination' in '/home/vagrant/katello/webpack/components/pf3Table/components'
19:21:19 webpack.1 | 
19:21:19 webpack.1 | ERROR in ../katello/webpack/scenes/RedHatRepositories/helpers.js
19:21:19 webpack.1 | Module not found: Error: Can't resolve 'foremanReact/components/Pagination' in '/home/vagrant/katello/webpack/scenes/RedHatRepositories'
19:21:19 webpack.1 | webpack: Failed to compile.
```

**Comment by jeremylenz** [comment]
@Ron-Lavi oddly, I don't see any of the JS test failures locally. maybe it's something to do with the Github action workflow that you changed?

**Comment by jeremylenz** [comment]
disregard the deleted comment, I _am_ seeing the JS test failures locally, 56 failed.
```
Configuration error:
    
    Could not locate module foremanReact/components/Pagination mapped as:
    /home/vagrant/foreman/webpack/assets/javascripts/react_app//components/Pagination.
    
    Please check your configuration for these entries:
    {
      "moduleNameMapper": {
        "/^foremanReact(.*)$/": "/home/vagrant/foreman/webpack/assets/javascripts/react_app/$1"
      },
      "resolver": null
    }

      4 | import { noop } from 'foremanReact/common/helpers';
      5 | import EmptyState from 'foremanReact/components/common/EmptyState';
    > 6 | import Pagination from 'foremanReact/components/Pagination';
```

**Comment by Ron-Lavi** [comment]
Fixed a few lines in Foreman context and the Pagination component so it won't fail in Katello's tests

**Comment by jeremylenz** [comment]
[test katello]

**Comment by Ron-Lavi** [comment]
The [core PR](https://github.com/theforeman/foreman/pull/8990) is approved now and ready,
I removed the GitHub action checkout of my Foreman commit as the test here was green.

**Comment by jeremylenz** [comment]
[test katello]

**Comment by jeremylenz** [comment]
[test katello]

**Review by jeremylenz — APPROVED** [review]
Code changes LGTM; looks to me like Katello was already consuming PaginationWrapper from core so there really aren't major changes on our side.

Clicked around tables in 
* new host details page
* new CV UI
* Red Hat Repositories page
* Subscriptions page
(did I miss any React pages with tables?)

Everything works fine. APJ :+1:

**Comment by jeremylenz** [comment]
@Ron-Lavi looks like 2 snapshots still need updating

**Comment by Andrewgdewar** [comment]
[test katello]

**Pull request merged**